src/file.cc

```diff
@@ -245,6 +248,9 @@ void write_buffer_to_fd(Buffer& buffer, int fd)
 
 void write_buffer_to_file(Buffer& buffer, StringView filename)
 {
+    if (buffer.flags() & Buffer::Flags::ReadOnly)
```

Comment:
This is called by `write_buffer`, so this means we wont be able to write buffer even if we specify the filename explicitely.